pkg/scheduler/admission/podnodeconstraints/admission.go

```diff
@@ -0,0 +1,175 @@
+package podnodeconstraints
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	admission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints/api"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeConstraints", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewPodNodeConstraints(pluginConfig), nil
+	})
+}
+
+// NewPodNodeConstraints creates a new admission plugin to prevent objects that contain pod templates
+// from containing node bindings by name or selector based on role permissions.
+func NewPodNodeConstraints(config *api.PodNodeConstraintsConfig) admission.Interface {
+	return &podNodeConstraints{
+		config:  config,
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+type podNodeConstraints struct {
+	*admission.Handler
+	client client.Interface
+	config *api.PodNodeConstraintsConfig
+}
+
+var _ = oadmission.WantsOpenshiftClient(&podNodeConstraints{})
+var _ = oadmission.Validator(&podNodeConstraints{})
+
+func readConfig(reader io.Reader) (*api.PodNodeConstraintsConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return &api.PodNodeConstraintsConfig{}, nil
```

Comment:
No, it should be a nil config so you know you're not enabled.  `ReadYAML` already does this for you.